srml/eth-backing/src/tests.rs

```diff
@@ -55,3 +61,191 @@ fn verify_parse_token_redeem_proof() {
+				header_hash: H256::from(hex!("14bf4b76a25a23ca1d625ff89673813548e138f84b511059695d801c0a7be578"))
+			};
+
+			let mixh = H256::from(hex!("3f3b1e56a051f395e9ee297a8bcd307ed7e328891d61eeb46b224dbdf710634a"));
+			let nonce = H64::from(hex!("87cce326ad070e33"));
+
+			let header = EthHeader {
+				parent_hash: H256::from(hex!("6ff4be9ac4f39a5e3886874bb939437b752e0c6f27803f9050b32c27f925a214")),
+				timestamp: 0x5df9feec,
+				number: 6988980,
+				author: EthAddress::from(hex!("635b4764d1939dfacd3a8014726159abc277becc")),
+				transactions_root: H256::from(hex!("3126f8c7133dff518b1d7dee4885a7179c9e68b37f792b2192663cd033961385")),
+				// sha3Uncles
+				uncles_hash: H256::from(hex!("1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347")),
+				extra_data: "de8302050a8f5061726974792d457468657265756d86312e33382e30826c69".from_hex().unwrap(),
+				state_root: H256::from(hex!("d90785981dfb161936c9d6cd5388da2ad98cf31717a0dfc7fce8b7dace0c8d07")),
+				receipts_root: H256::from(hex!("73f38c8ee57395ba8c46d856e33e3042738143d7db02f61881b714069b58982a")),
+				log_bloom: Bloom::from_str("04000008000000401000000080002000000011000001001000000000025000000000000100000000000000000000000000000000000000000800000000002000000000000000000010000048000000a40000000020400000000020000080800000000000420200020000000000000800080000000000080001008010000000001000000400000000000000000000001000000420000400000041000000300000008000001000000000000100000000000801008000000200000000000000000012000002000000000000800000848000000481000100000000000100420020001210004000040080004000000000100000400000018000004000040000000000").unwrap(),
+				gas_used: 0x141bd0.into(),
+				gas_limit: 0x7a121d.into(),
+				difficulty: 0x745523c7_u64.into(),
+				seal: vec![rlp::encode(&mixh), rlp::encode(&nonce)],
+				hash: Some(H256::from(hex!("14bf4b76a25a23ca1d625ff89673813548e138f84b511059695d801c0a7be578"))),
+			};
+
+			// totalDifficulty
+			assert_ok!(EthRelay::init_genesis_header(&header, 0x68e58ae1c31caf_u64));
+
+		 	let expect_account_id = <Test as Trait>::DetermineAccountId::account_id_for(&hex!("2a92ae5b41feba5ee68a61449c557efa9e3b894a6461c058ec2de45429adb44546")).ok().unwrap();
+			
+			let id1 = AccountId32::from([0; 32]);
+			
+			let _ = Ring::deposit_creating(&expect_account_id, 1);
+			assert_ok!(
+				staking::Call::<Test>::bond(
+					AccountId32::from([1; 32]),
+					StakingBalances::Ring(1),
+					RewardDestination::Controller,
+					0).dispatch(Origin::signed(expect_account_id.clone()))
+			); 	
+		 	assert_ok!(EthBacking::redeem_deposit(Origin::signed(id1.clone()), proof_record.clone()));
+
+		 	assert_eq!(Ring::free_balance(&expect_account_id), 1234000000000 + 1);
```

Comment:
TODO: Assert deposit Item and active ring

Reply:
Fixed in https://github.com/darwinia-network/darwinia/pull/172/commits/07694a353093cad2723c3f08001d702c4fe8664b